server/src/main/java/io/unitycatalog/server/persist/ExternalLocationRepository.java

```diff
@@ -0,0 +1,195 @@
+      ExternalLocationDAO externalLocationDAO =
+          ExternalLocationDAO.builder()
+              .id(externalLocationId)
+              .name(createExternalLocation.getName())
+              .url(createExternalLocation.getUrl())
+              .comment(createExternalLocation.getComment())
+              .owner(callerId)
+              .accessPoint(createExternalLocation.getAccessPoint())
+              .credentialId(storageCredentialDAO.getId())
+              .createdAt(new Date())
+              .createdBy(callerId)
+              .build();
+      Transaction tx = session.beginTransaction();
+      try {
+        if (getExternalLocationDAO(session, createExternalLocation.getName()) != null) {
+          throw new BaseException(
+              ErrorCode.ALREADY_EXISTS,
+              "External location already exists: " + createExternalLocation.getName());
+        }
+        session.persist(externalLocationDAO);
+        tx.commit();
+        return externalLocationDAO;
+      } catch (Exception e) {
+        tx.rollback();
+        throw e;
+      }
+    }
+  }
+
+  public ExternalLocationInfo getExternalLocation(String name) {
+    try (Session session = sessionFactory.openSession()) {
+      session.setDefaultReadOnly(true);
+      Transaction tx = session.beginTransaction();
+      try {
+        ExternalLocationDAO externalLocationDAO = getExternalLocationDAO(session, name);
+        if (externalLocationDAO == null) {
+          throw new BaseException(ErrorCode.NOT_FOUND, "External location not found: " + name);
+        }
+        tx.commit();
+        return externalLocationDAO.toExternalLocationInfo();
+      } catch (Exception e) {
+        tx.rollback();
+        throw e;
+      }
+    }
+  }
+
+  public ExternalLocationDAO getExternalLocationDAO(Session session, String name) {
+    Query<ExternalLocationDAO> query =
+        session.createQuery(
+            "FROM ExternalLocationDAO WHERE name = :value", ExternalLocationDAO.class);
+    query.setParameter("value", name);
+    query.setMaxResults(1);
+    return query.uniqueResult();
+  }
+
+  public ListExternalLocationsResponse listExternalLocations(
+      Optional<Integer> maxResults, Optional<String> pageToken) {
+    try (Session session = sessionFactory.openSession()) {
+      session.setDefaultReadOnly(true);
+      Transaction tx = session.beginTransaction();
+      try {
+        List<ExternalLocationDAO> daoList =
+            LISTING_HELPER.listEntity(session, maxResults, pageToken, null);
+        String nextPageToken = LISTING_HELPER.getNextPageToken(daoList, maxResults);
+        List<ExternalLocationInfo> results = new ArrayList<>();
+        for (ExternalLocationDAO dao : daoList) {
+          results.add(dao.toExternalLocationInfo());
+        }
+        tx.commit();
+        return new ListExternalLocationsResponse()
+            .externalLocations(results)
+            .nextPageToken(nextPageToken);
+      } catch (Exception e) {
+        tx.rollback();
```

Comment:
please add logging!
its very very important to log errors even if you are rethrowing them.